docking.js

```diff
@@ -1512,6 +1516,60 @@ const DockedDash = new Lang.Class({
         }, this);
 
         this._hotKeysEnabled = false;
+    },
+
+    _optionalNumberOverlay: function() {
+        if (this._settings.get_boolean('hot-keys'))
+            this._enableNumberOverlay();
+
+        this._signalsHandler.add([
+            this._settings,
+            'changed::hot-keys',
+            Lang.bind(this, function() {
+                    if (this._settings.get_boolean('hot-keys'))
+                        this._enableNumberOverlay();
+                    else
+                        this._disableNumberOverlay();
+            })
+        ]);
+    },
+
+    _enableNumberOverlay: function() {
+        this.keymap = Gdk.Keymap.get_default();
+
+        this._signalsHandler.removeWithLabel('number-overlay');
+        this._signalsHandler.addWithLabel('number-overlay', [
+            this.keymap,
+            'state_changed',
+            Lang.bind(this, function() {
+                    let state = this.keymap.get_modifier_state();
```

Comment:
I actually don't like 100% this approach. I tried to use the signal `key-press-event` as it's done here (https://extensions.gnome.org/extension/10/windownavigator/), but I couldn't get it to work.
Instead, I took this idea from: https://extensions.gnome.org/extension/975/keyboard-modifiers-status/